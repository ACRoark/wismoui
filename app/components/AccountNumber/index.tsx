import * as React from 'react';

import formatAccountNumber from './formatAccountNumber';

interface IAccountNumberProps {
  value: string;
}

const AccountNumber: React.FC<IAccountNumberProps> = (props: IAccountNumberProps): React.ReactElement => {
  const { value } = props;

  const formattedAccountNumber = formatAccountNumber(value);

  return <span>{formattedAccountNumber}</span>;
};

export default AccountNumber;
