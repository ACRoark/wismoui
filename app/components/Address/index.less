@import (reference) '../../styles/themes/index.less';

@address-cls-prefix: ~'@{dte-prefix}-address';

.@{address-cls-prefix} {
  display: inline;
  font-style: normal;
  line-height: 1.375rem;
  white-space: pre-line;
}
