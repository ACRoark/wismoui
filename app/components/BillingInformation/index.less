@import (reference) '../../styles/themes/index.less';

@billing-information-cls-prefix: ~'@{dte-prefix}-billing-information';

.@{billing-information-cls-prefix} {
  background-color: @dte-gray-200;
  margin-bottom: 28px;
  padding: 20px;

  > a {
    &:link {
      font-weight: @dte-wismo-font-weight-semi-bold;
    }
    &:visited {
      color: @dte-blue-light;
    }
    &:hover {
      color: @dte-interactive-hover;
    }
  }
}
