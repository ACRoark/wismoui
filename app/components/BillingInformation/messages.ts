import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.billing-information';

export default defineMessages({
  authenticatedUsers: {
    id: `${scope}.authenticated-users`,
  },
  billingInformationLabel: {
    id: `${scope}.billing-information-label`,
  },
  guestPayLink: {
    id: `${scope}.guest-pay-link`,
  },
  MIMO_START: {
    id: `${scope}.MIMO_START`,
  },
  MIMO_STOP: {
    id: `${scope}.MIMO_STOP`,
  },
  signInLink: {
    id: `${scope}.sign-in-link`,
  },
});
