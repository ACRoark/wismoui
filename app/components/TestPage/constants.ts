import IOrderRequest from 'types/IOrderRequest';
import IProduct from '../../types/IProduct';

export const fakeProductList: IProduct[] = [
  {
    productType: 'GCR_A_H',
    serviceDate: '2020-04-22',
  },
  {
    productType: 'EFR_D1',
    serviceDate: '2020-04-22',
  },
  {
    productType: 'EFR_D1_1',
    serviceDate: '2020-04-22',
  },
];

export const fakeServiceRequest: IOrderRequest = {
  accountNumber: '111122233334',
  address: {
    city: 'Bloomfield',
    line1: '123 Main',
    line2: null,
    state: 'MI',
    zip: '48301',
  },
  contactPhoneNumber: '7348675309',
  orderRequestStatusUpdates: [
    {
      createdAtDate: '20200319',
      createdAtTime: '131530',
      orderRequestStatus: 'REQUESTED',
    },
  ],
  orderRequestType: 'MIMO_START',
  premiseId: 'premise123',
  products: [
    {
      productType: 'GCR_A_H',
      serviceDate: '2020-04-22',
    },
    {
      productType: 'EFR_D1',
      serviceDate: '2020-04-22',
    },
    {
      productType: 'EFR_D1_1',
      serviceDate: '2020-04-22',
    },
  ],
  serviceOrderEvents: [
    {
      appointment: {
        date: '2020-04-17',
        slotType: 'AM',
      },
      latestStatus: {
          createdAtDate: '20200305',
          createdAtTime: '223443',
          serviceOrderEventStatus: 'COMPLETED',
        },
        product: 'EFR_D1',
      serviceOrderCategory: null,
      serviceOrderId: '123456789',
    },
  ],
  verificationEvents: [
    {
      verificationCategory: 'R975',
      verificationEventId: '123123',
      verificationEventStatusUpdates: [
        {
          verificationCode: 'AD',
          verificationEventStatus: 'COMPLETED',
          createdAtDate: '20200319',
          createdAtTime: '131530',
        },
      ],
    },
  ],
  wantDate: '2020-04-22',
};

export const fakeStartServiceRequest: IOrderRequest = {
  ...fakeServiceRequest,
  orderRequestType: 'MIMO_START',
};

export const fakeStopServiceRequest: IOrderRequest = {
  ...fakeServiceRequest,
  orderRequestType: 'MIMO_STOP',
};
