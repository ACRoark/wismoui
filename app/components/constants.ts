import IApplicationConfig from 'types/IApplicationConfig';

export const PageTitles = Object.freeze({
  ORDER_STATUS_PAGE: 'Track Your Order | DTE Energy',
  SEARCH_PAGE: 'Order Tracker | DTE Energy',
  SUMMARY_PAGE: 'Track Your Order | DTE Energy',
});

export const testConfig: IApplicationConfig = {
  flags: {
    canChangeLanguage: false,
    developerMode: false,
  },
  gigyaSiteKey: 'foo',
  googleTagManagerConfig: {
    auth: 'p3zyt1YCX84XB82sOrjMGw', // '',
    id: 'GTM-K2WP588', // 'GTM-PBB6QZ4',
    preview: 'env-1', // '',
  },
  loading: true,
  recaptchaSitekey: '6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI',
  urls: {
    assistanceURL: 'https://www.dteenergy.com',
    backToDTE: 'https://www.dteenergy.com',
    backToMIMO: 'https://www.dteenergy.com',
    contactUs: 'https://www.contact.us.com',
    createAccount: 'https://www.create.account.com',
    decryptTokenApi: 'https://web-test.dteenergy.com/api/decryptToken?detokenizeValue=',
    documentSubmission: 'https://web-test.dteenergy.com/wps/wcm/connect/dte-web/quicklinks/web-form?detokenizeValue=',
    dynatraceScript:
      'https://dynatrace.dteenergy.com:9999/jstag/managed/98834406-1ae1-4257-992c-5b3c5f828584/968d8c7df07fe18c_complete.js',
    editPhoneNumber: 'https://www.edit.phone.number.com',
    editWantDate: 'https://www.edit.service.date.com',
    encryptTokenApi: 'https://web-test.dteenergy.com/api/encryptToken',
    findOrder: 'https://firebase.or.jsonserver/orders/{orderNumber}?name={name}',
    getOrder: 'https://firebase.or.jsonserver/orders/{orderNumber}',
    guestPay: 'https://www.guest.pay.com',
    headerLogo: 'https://www.header.logo.com',
    privacyPolicy: 'https://www.privacy.policy.com',
    searchOrders: 'https://firebase.or.jsonserver/orders',
    signIn: 'https://www.sign.in.com',
    termsAndConditions: 'https://www.terms.and.conditions.com',
  },
};
