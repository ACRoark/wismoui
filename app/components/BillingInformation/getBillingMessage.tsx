import React from 'react';
import { FormattedMessage } from 'react-intl';

import IUrlConfig from 'types/IUrlConfig';

import messages from './messages';

const getBillingMessage = (
  urls: IUrlConfig,
  requestType: string,
  authenticated: boolean,
): React.ReactElement => {

  const GuestPayLink = (
    <a href={urls.guestPay}>
      <FormattedMessage {...messages.guestPayLink} />
    </a>
  );

  const SignInLink = (
    <a href={urls.signIn}>
      <FormattedMessage {...messages.signInLink} />
    </a>
  );

  if (authenticated) {
    return (
      <FormattedMessage
        {...messages.authenticatedUsers}
        values={{
          a: (chunks: React.ReactNode): React.ReactNode => (
            <a href={urls.signIn} target="_blank">
              {chunks}
            </a>
          ),
        }}
      />
    );
  }

  return (
    <FormattedMessage
      {...messages[requestType]}
      values={{
        guestPay: GuestPayLink,
        signIn: SignInLink,
      }}
    />
  );
};

export default getBillingMessage;
