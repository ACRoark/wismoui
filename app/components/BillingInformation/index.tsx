import React from 'react';

import useConfig from 'hooks/useConfig';
import { userIsAuthenticated } from 'security/utils';

import getBillingMessage from './getBillingMessage';
import './index.less';

interface IBillingInformationProps {
  requestType: string;
}

const BillingInformation: React.FC<IBillingInformationProps> = (
  props: IBillingInformationProps,
): React.ReactElement => {
  const { requestType } = props;

  const isAuthenticated = userIsAuthenticated();
  const { urls } = useConfig();

  const BillingMessage = getBillingMessage(urls, requestType, isAuthenticated);

  return <div className="dte-wismo-billing-information">{BillingMessage}</div>;
};

export default BillingInformation;
