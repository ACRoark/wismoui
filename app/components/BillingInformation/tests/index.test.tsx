import React from 'react';
import { Provider } from 'react-redux';

import { SUPPORTED_LOCALES } from 'locales';

import { testConfig } from 'components/constants';
import useConfig from 'hooks/useConfig';
import userIsAuthenticated from 'security/utils/userIsAuthenticated';
import { createMockStore, createSnapshotWithIntl, renderWithIntl } from 'testing/utils';

import BillingInformation from '..';

jest.mock('security/utils/userIsAuthenticated');
jest.mock('hooks/useConfig');

const mockAuth = userIsAuthenticated as jest.MockedFunction<typeof userIsAuthenticated>;
const mockConfig = useConfig as jest.MockedFunction<typeof useConfig>;

const requestType = ['MIMO_START', 'MIMO_STOP'];
const store = createMockStore();

const expectBillingInformationToMatchSnapshot = (authValue: boolean, language: string, orderRequestType: string): void => {
  mockAuth.mockReturnValue(authValue);
  mockConfig.mockReturnValue({ ...testConfig });

  const tree = createSnapshotWithIntl(<Provider store={store}><BillingInformation
    requestType={orderRequestType} /></Provider>, language);

  expect(tree).toMatchSnapshot();
};

describe('<BillingInformation/>', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    requestType.forEach((orderRequestType: string): void => {

      it(`should not log errors in console when rendered in ${language} with requestType of ${orderRequestType}`, (): void => {
        const spy = jest.spyOn(global.console, 'error');

        mockAuth.mockReturnValue(false);
        mockConfig.mockReturnValue({ ...testConfig });

        renderWithIntl(<Provider store={store}><BillingInformation
          requestType={orderRequestType} /></Provider>, language);

        expect(spy).not.toHaveBeenCalled();
      });

      it(`should render correctly when the current language is ${language}, and requestType is ${orderRequestType}, and user is not authenticated`, (): void =>
        expectBillingInformationToMatchSnapshot(false, language, orderRequestType));

      it(`should render correctly when the current language is ${language}, and requestType is ${orderRequestType}, and user is authenticated`, (): void =>
        expectBillingInformationToMatchSnapshot(true, language, orderRequestType));
    });
  });
});
