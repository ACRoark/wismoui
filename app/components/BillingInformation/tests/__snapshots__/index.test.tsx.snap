// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BillingInformation/> should render correctly when the current language is en, and requestType is MIMO_START, and user is authenticated 1`] = `
<div
  className="dte-wismo-billing-information"
>
  Your bill is ready to view 
  <a
    href="https://www.sign.in.com"
    target="_blank"
  >
    online
  </a>
  .
</div>
`;

exports[`<BillingInformation/> should render correctly when the current language is en, and requestType is MIMO_START, and user is not authenticated 1`] = `
<div
  className="dte-wismo-billing-information"
>
  Your bill is ready to view online. Please 
  <a
    href="https://www.sign.in.com"
  >
    sign in
  </a>
   to make your payment or use our 
  <a
    href="https://www.guest.pay.com"
  >
    Guest Pay
  </a>
   option.
</div>
`;

exports[`<BillingInformation/> should render correctly when the current language is en, and requestType is MIMO_STOP, and user is authenticated 1`] = `
<div
  className="dte-wismo-billing-information"
>
  Your bill is ready to view 
  <a
    href="https://www.sign.in.com"
    target="_blank"
  >
    online
  </a>
  .
</div>
`;

exports[`<BillingInformation/> should render correctly when the current language is en, and requestType is MIMO_STOP, and user is not authenticated 1`] = `
<div
  className="dte-wismo-billing-information"
>
  Your bill is ready to view online. Please 
  <a
    href="https://www.sign.in.com"
  >
    sign in
  </a>
   to make your payment or use our 
  <a
    href="https://www.guest.pay.com"
  >
    Guest Pay
  </a>
   option.
</div>
`;

exports[`<BillingInformation/> should render correctly when the current language is es, and requestType is MIMO_START, and user is authenticated 1`] = `
<div
  className="dte-wismo-billing-information"
>
  Su factura está lista para ver 
  <a
    href="https://www.sign.in.com"
    target="_blank"
  >
    en línea
  </a>
  .
</div>
`;

exports[`<BillingInformation/> should render correctly when the current language is es, and requestType is MIMO_START, and user is not authenticated 1`] = `
<div
  className="dte-wismo-billing-information"
>
  Su factura está lista para ver en línea. 
  <a
    href="https://www.sign.in.com"
  >
    Inicie sesión
  </a>
   para realizar su pago o use nuestra opción 
  <a
    href="https://www.guest.pay.com"
  >
    Pago de Invitado
  </a>
</div>
`;

exports[`<BillingInformation/> should render correctly when the current language is es, and requestType is MIMO_STOP, and user is authenticated 1`] = `
<div
  className="dte-wismo-billing-information"
>
  Su factura está lista para ver 
  <a
    href="https://www.sign.in.com"
    target="_blank"
  >
    en línea
  </a>
  .
</div>
`;

exports[`<BillingInformation/> should render correctly when the current language is es, and requestType is MIMO_STOP, and user is not authenticated 1`] = `
<div
  className="dte-wismo-billing-information"
>
  Su factura está lista para ver en línea. 
  <a
    href="https://www.sign.in.com"
  >
    Inicie sesión
  </a>
   para realizar su pago o use nuestra opción 
  <a
    href="https://www.guest.pay.com"
  >
    Pago de Invitado
  </a>
</div>
`;
